src/map/map.cpp

```diff
@@ -194,6 +194,507 @@ struct s_generator_options {
+		const auto &jobNode = node["RestrictedJobs"];
+
+		for (const auto &it : jobNode) {
+			std::string job_name;
+			c4::from_chars(it.key(), &job_name);
+			std::string job_name_constant = "JOB_" + job_name;
+			int64 job_id;
+
+			if (!script_get_constant(job_name_constant.c_str(), &job_id)) {
+				this->invalidWarning(jobNode, "Job %s does not exist.\n", job_name.c_str());
+				continue;
+			}
+
+			uint16 group_lv;
+
+			if (!this->asUInt16(jobNode, job_name, group_lv))
+				continue;
+
+			if (group_lv > 100) {
+				this->invalidWarning(jobNode, "Job Group Level can not be above 100, capping to 100.\n");
+				group_lv = 100;
+			}
+
+			if (group_lv == 0)
+				zone->restricted_jobs.erase(static_cast<uint32>(job_id));
+			else
+				zone->restricted_jobs[static_cast<uint32>(job_id)] = group_lv;
+		}
+	}
+
+	if (this->nodeExists(node, "Maps")) {
+		const auto &mapNode = node["Maps"];
+
+		for (const auto &it : mapNode) {
+			std::string map_name;
+			c4::from_chars(it.key(), &map_name);
+			int16 mapidx = mapindex_name2idx(map_name.c_str(), nullptr);
+
+			if (mapidx == 0) {
+				this->invalidWarning(mapNode, "Map %s does not exist.\n", map_name.c_str());
+				continue;
+			}
+
+			bool enabled;
+
+			if (!this->asBool(mapNode, map_name, enabled))
+				continue;
+
+			int16 map_id = map_mapname2mapid(map_name.c_str());
+
+			if (enabled) {
+				if (util::vector_exists(zone->maps, map_id)) {
+					this->invalidWarning(mapNode, "Map %s is already part of this zone.\n", map_name.c_str());
+					continue;
+				}
+
+				zone->maps.push_back(map_id);
+			} else {
+				if (!util::vector_erase_if_exists(zone->maps, map_id)) {
+					this->invalidWarning(mapNode, "Map %s is not part of this zone.\n", map_name.c_str());
+					continue;
+				}
+			}
+		}
+	}
+
+	if (this->nodeExists(node, "Mapflags")) {
+		const auto &mapflagNode = node["Mapflags"];
+
+		for (const auto &it : mapflagNode) {
+			std::string flag_name;
+
+			if (!this->asString(it, "Flag", flag_name))
+				return 0;
+
+			std::string flag_name_constant = "MF_" + flag_name;
+			int64 flag;
+
+			if (!script_get_constant(flag_name_constant.c_str(), &flag)) {
+				this->invalidWarning(mapflagNode, "Mapflag %s does not exist.\n", flag_name.c_str());
+				continue;
+			}
+
+			std::string value;
+
+			if (this->nodeExists(it, "Value")) {
+				if (!this->asString(it, "Value", value))
+					continue;
+			} else
+				value = "1";
+
+			bool clear = false;
+
+			if (this->nodeExists(it, "Clear")) {
+				if (!this->asBool(it, "Clear", clear))
+					continue;
+			}
+
+			if (!clear) {
+				zone->mapflags.insert({ std::pair<int16, std::string>(static_cast<int16>(flag), value) });
+			} else {
+				// Get all mapflag keys that match
+				auto mapflagit = zone->mapflags.equal_range(static_cast<int16>(flag));
+
+				// Iterate over the mapflags to get their values
+				for (std::multimap<int16, std::string>::iterator it = mapflagit.first; it != mapflagit.second; ++it) {
```

Comment:
```suggestion
				for (auto it = mapflagit.first; it != mapflagit.second; ++it) {
```
But I think "it" shadows the loop variable of the outer loop here.